eks/cluster-loader/clusterloader2/addon.go

```diff
@@ -59,6 +63,22 @@ func (c *ClusterLoader) Apply() (err error) {
 			Ready:     true,
 		},
 	}
+
+	//Wait for job to complete -- 2 hours because larger tests take a very long time.
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Hour)
+	job := &v1.Job{}
+	for job.Status.Succeeded < 1 {
+		job, err = c.K8sClient.KubernetesClientSet().
+			BatchV1().
+			Jobs("clusterloader2").
+			Get(ctx, "clusterloader2", metav1.GetOptions{})
+		if err != nil {
```

Comment:
We will need to improve this in the future. Right now your test will exit if there's a brief networking issue, which happens more often than you might think. You could explore https://godoc.org/github.com/cenkalti/backoff.

Generally, we want our processes to be tolerant to brief networking issues, while still failing fast if there's a true problem. Identifying this can be challenging, but backoff retry is a good approach.

Reply:
+1